### DISCUSSION TITLE
How to write your code to be better rectified

### DISCUSSION THREAD

**Discussion by sabbelasichon** [sabbelasichon/typo3-rector#2147] [Ideas]
Add hints how to write your code to be better rectified

**Comment by DanielSiepmann** [comment]
Rector is able to migrate so much code.
But sometimes it is possible to solve problems in many ways. Rector would need to know all of them.
I would suggest providing a small documentation about how to write code to enable rector to migrate that part. That would reduce bloat from this project and waste time writing rectors.

I've written a small blog post about preparing code for DI and Events: https://daniel-siepmann.de/prepare-legacy-code-for-upcoming-typo3-versions.html
That could be a starting point. E.g. https://github.com/sabbelasichon/typo3-rector/blob/master/src/Rector/Experimental/OptionalConstructorToHardRequirementRector.php should work with the dependency injection example.

Providing such documentation allows developers to write code, which can be migrated via rector. They don't need to worry and care about the update and other solutions.

**Reply by sabbelasichon** [reply]
@DanielSiepmann Would be nice if you could do that and spread the word. Thank you.
